t/09-job_clone.t

```diff
@@ -19,12 +19,30 @@ use Test::Most;
 use FindBin;
 use lib "$FindBin::Bin/lib";
 use OpenQA::Utils;
+use Mojo::File 'tempdir';
+use OpenQA::Script::CloneJob;
 use OpenQA::Test::Database;
 use Test::Mojo;
 use Test::Warnings ':report_warnings';
 
 OpenQA::Test::Database->new->create();
 my $t = Test::Mojo->new('OpenQA::WebAPI');
+# XXX: https://github.com/kraih/mojo/issues/598
+my $app = $t->app;
+$t->ua(
+    OpenQA::Client->new(apikey => 'PERCIVALKEY02', apisecret => 'PERCIVALSECRET02')->ioloop(Mojo::IOLoop->singleton));
+$t->app($app);
+my $dir = tempdir("/tmp/$FindBin::Script-XXXX")->make_path;
+
+my $daemon = Mojo::Server::Daemon->new(listen => ['http://127.0.0.1']);
+$daemon->build_app('OpenQA::WebAPI');
+my $port = $daemon->start->ports->[0];
+my $host = "http://127.0.0.1:$port";
```

Comment:
IMHO a bit too much copy-pasted boiler-plate code. I am fine with the functional change but could you please try to extract the common code into helper functions? E.g. move this into one or two helper functions in OpenQA::Test::Case

Reply:
I changed it to use `create_webapi` although it's not less code.

Might be nice to also wrap the `Test::Mojo + OpenQA::Client` pattern but I'd rather do that separately since it's exactly what we do in everywhere.